### PR TITLE
Epic homepage navigation

### PR DESCRIPTION
**Pull request by lmiotti**
### Summary
Homepage layout created with news and profile tabs. If a tab is selected its icon and inscription is turned to green and the other one is turned to grey/black

### Screenshot
image

### Trello Card
https://trello.com/c/CJrsHAb3/17-u4-home-page-tabs-de-navegaci%C3%B3n

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -91,6 +91,11 @@ buildscript {
     ext.wolmo_version = 'v1.3.1'
 }
 
+repositories {
+    jcenter()
+}
+
+
```
Comment (alvarez-fabian) [inline app/build.gradle]: Extra empty line :smile:

File: app/build.gradle [hunk 2]
```diff
@@ -101,6 +106,7 @@ dependencies {
     // Android support libraries
     implementation "com.android.support:support-compat:$support_library_version"
     implementation "com.android.support:recyclerview-v7:$support_library_version"
+    implementation 'com.android.support:recyclerview-v7:27.1.0'
```
Comment (alvarez-fabian) [inline app/build.gradle]: I don't think we should add the same library twice. Remove this line :+1:

File: app/build.gradle [hunk 3]
```diff
@@ -116,4 +122,5 @@ dependencies {
     testImplementation 'org.mockito:mockito-core:1.10.19'
     testImplementation 'org.hamcrest:hamcrest-library:1.3'
     testImplementation "com.squareup.leakcanary:leakcanary-android-no-op:$leak_canary_version"
+    testCompile "org.mockito:mockito-core:2.+"
```
Comment (alvarez-fabian) [inline app/build.gradle]: I think you already changed this in another branch, but you can remove if you want.

File: app/src/main/java/ar/com/wolox/android/training/ui/home/HomeFragment.java [hunk 4]
```diff
@@ -1,28 +1,88 @@
...
+import butterknife.OnClick;
+import butterknife.OnItemSelected;
 
 public class HomeFragment extends WolmoFragment<HomePresenter> implements HomeView {
 
+    @BindView(R.id.home_tab_layout) TabLayout mTabLayout;
+    @BindView(R.id.home_pager) ViewPager mPager;
+    @BindView(R.id.home_recycler_view) RecyclerView mRecyclerView;
+
+
```
Comment (alvarez-fabian) [inline app/src/main/java/ar/com/wolox/android/training/ui/home/HomeFragment.java]: Try not to leave more than 1 empty line between methods to keep the code clean :smile:

File: app/src/main/java/ar/com/wolox/android/training/ui/home/HomeFragment.java [hunk 5]
```diff
@@ -1,28 +1,88 @@
...
         fragment.setArguments(args);
         return fragment;
     }
 
+
     @Override
-    public int layout() {
-        return R.layout.fragment_home;
-    }
+    public int layout() { return R.layout.fragment_home;}
```
Comment (alvarez-fabian) [inline app/src/main/java/ar/com/wolox/android/training/ui/home/HomeFragment.java]: Only join curly braces when the method is empty, as example:
`public void setListeners() {}`
But if you have code please move it to a new line :smiley:

File: app/src/main/java/ar/com/wolox/android/training/ui/home/HomeFragment.java [hunk 6]
```diff
@@ -1,28 +1,88 @@
...
 
     @Override
     public void setUi(View v) {
         super.setUi(v);
+
+        //News Tab
+        TabLayout.Tab NewsTab = mTabLayout.newTab();
+        NewsTab.setText("News");
```
Comment (alvarez-fabian) [inline app/src/main/java/ar/com/wolox/android/training/ui/home/HomeFragment.java]: You should extract this string an put it in `strings.xml` and then reference it here like `getString(R.string.news_title)` or something like that.

File: app/src/main/java/ar/com/wolox/android/training/ui/home/HomeFragment.java [hunk 7]
```diff
@@ -1,28 +1,88 @@
...
+
+        //Profile Tab
+        TabLayout.Tab ProfileTab = mTabLayout.newTab();
+        ProfileTab.setText("Profile");
```
Comment (alvarez-fabian) [inline app/src/main/java/ar/com/wolox/android/training/ui/home/HomeFragment.java]: Same as above :smile:

File: app/src/main/java/ar/com/wolox/android/training/ui/home/HomeFragment.java [hunk 8]
```diff
@@ -1,28 +1,88 @@
...
+
+        mTabLayout.setOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
+            @Override
+            public void onTabSelected(TabLayout.Tab tab) {
+                if(tab == NewsTab) {
+                    NewsTab.setIcon(R.drawable.ic_news_list_on);
```
Comment (alvarez-fabian) [inline app/src/main/java/ar/com/wolox/android/training/ui/home/HomeFragment.java]: You can accomplish the same with a `selector` and keep the code clean. Please ask me or @DemianIac how to do that :+1:

File: app/src/main/java/ar/com/wolox/android/training/ui/home/HomeFragment.java [hunk 9]
```diff
@@ -1,28 +1,88 @@
...
+                }
+            }
+
+            @Override
+            public void onTabUnselected(TabLayout.Tab tab) {
+            }
+
+            @Override
+            public void onTabReselected(TabLayout.Tab tab) {
+
```
Comment (alvarez-fabian) [inline app/src/main/java/ar/com/wolox/android/training/ui/home/HomeFragment.java]: Remember to remove empty lines :smile:

File: app/src/main/java/ar/com/wolox/android/training/ui/home/HomeFragment.java [hunk 10]
```diff
@@ -1,28 +1,88 @@
...
+            }
+        });
+        
+        mTabLayout.addTab(NewsTab, true);
+        mTabLayout.addTab(ProfileTab, false);
+        mTabLayout.setTabTextColors(Color.parseColor("#a5a8a9"), Color.parseColor("#8DC63F"));
```
Comment (alvarez-fabian) [inline app/src/main/java/ar/com/wolox/android/training/ui/home/HomeFragment.java]: Move these colors to `colors.xml` and import them here.

File: app/src/main/java/ar/com/wolox/android/training/ui/home/HomeFragment.java [hunk 11]
```diff
@@ -1,28 +1,88 @@
...
+
+        //News Tab
+        TabLayout.Tab NewsTab = mTabLayout.newTab();
+        NewsTab.setText("News");
+
+        //Profile Tab
+        TabLayout.Tab ProfileTab = mTabLayout.newTab();
+        ProfileTab.setText("Profile");
+
+        mTabLayout.setOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
```
Comment (alvarez-fabian) [inline app/src/main/java/ar/com/wolox/android/training/ui/home/HomeFragment.java]: You can move this to `setListeners()` method to keep this method more organized :ok_hand:

File: app/src/main/res/layout/fragment_home.xml [hunk 12]
```diff
@@ -1,12 +1,55 @@
...
-    xmlns:android="http://schemas.android.com/apk/res/android"
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
     android:layout_width="match_parent"
-    android:layout_height="match_parent">
-    <TextView
+    android:layout_height="match_parent"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:gravity="center"
+    android:orientation="vertical"
+    android:padding="1dp"
```
Comment (alvarez-fabian) [inline app/src/main/res/layout/fragment_home.xml]: Hardcoded dimen. You can move this to `dimens.xml`. And then use `@dimen/padding_smallest` or something like that.

File: app/src/main/res/layout/fragment_home.xml [hunk 13]
```diff
@@ -1,12 +1,55 @@
 <?xml version="1.0" encoding="utf-8"?>
-<LinearLayout
-    xmlns:android="http://schemas.android.com/apk/res/android"
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
     android:layout_width="match_parent"
-    android:layout_height="match_parent">
-    <TextView
+    android:layout_height="match_parent"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
```
Comment (alvarez-fabian) [inline app/src/main/res/layout/fragment_home.xml]: Try to keep this order of attributes in the XML:
```
xmlns:******
android:layout_width
android:layout_height
android:padding**
the rest in alphabetical order (not required)
```

File: app/src/main/res/layout/fragment_home.xml [hunk 14]
```diff
@@ -1,12 +1,55 @@
...
+    android:gravity="center"
+    android:orientation="vertical"
+    android:padding="1dp"
+    android:clipChildren="false">
+
+    <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (alvarez-fabian) [inline app/src/main/res/layout/fragment_home.xml]: You don't need to import the namespace here. Move it to the root parent :smile:

File: app/src/main/res/layout/fragment_home.xml [hunk 15]
```diff
@@ -1,12 +1,55 @@
...
+
+    <android.support.v7.widget.RecyclerView
+        android:id="@+id/home_recycler_view"
+        android:scrollbars="vertical"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"/>
+
 
-</LinearLayout>
+    <!--<View-->
```
Comment (alvarez-fabian) [inline app/src/main/res/layout/fragment_home.xml]: Remove commented view :smile:

File: app/src/main/res/layout/fragment_home.xml [hunk 16]
```diff
@@ -1,12 +1,55 @@
...
+        <!--android:layout_width="match_parent"-->
+        <!--android:layout_height="0dp"-->
+        <!--android:layout_weight="1"/>-->
 
+</LinearLayout>
```
Comment (alvarez-fabian) [inline app/src/main/res/layout/fragment_home.xml]: Remember to not to have more than 2 empty lines :smile: I comment it here to no mention it in all the file :ok_hand:

**Review by alvarez-fabian — CHANGES_REQUESTED** [review]
Nice work! Keep it up.
Mostly style comments, nothing big. Assign the Pull Request to me when you are done fixing it. :muscle:

File: app/src/main/java/ar/com/wolox/android/training/ui/home/HomeFragment.java (hunk 6 above)
Comment (lmiotti) [reply app/src/main/java/ar/com/wolox/android/training/ui/home/HomeFragment.java]: "News" reference was solved in a new branch!

File: app/src/main/java/ar/com/wolox/android/training/ui/home/HomeFragment.java (hunk 7 above)
Comment (lmiotti) [reply app/src/main/java/ar/com/wolox/android/training/ui/home/HomeFragment.java]: "Profile" string too!

**Comment by lmiotti** [comment]
Empty and commented lines were erased, extra libraries were removed, strings were referenced to strings.xml, colors referenced to colors.xml, setListeners() method created, harcoded dimens were solved and two selectors were implemented ( one for news tab icon and the other for profile tab icon).

File: app/src/main/java/ar/com/wolox/android/training/ui/home/HomeFragment.java [hunk 17]
```diff
@@ -30,40 +32,29 @@
...
             @Override
             public void onTabSelected(TabLayout.Tab tab) {
-                if(tab == NewsTab) {
-                    NewsTab.setIcon(R.drawable.ic_news_list_on);
-                    ProfileTab.setIcon(R.drawable.ic_profile_off);
-                }else {
-                    NewsTab.setIcon(R.drawable.ic_news_list_off);
-                    ProfileTab.setIcon(R.drawable.ic_profile_on);
-                }
+//                if(tab == NewsTab) {
```
Comment (alvarez-fabian) [inline app/src/main/java/ar/com/wolox/android/training/ui/home/HomeFragment.java]: If you don't need this block of code, just remove it :smile:

File: app/src/test/java/ar/com/wolox/android/training/ui/login/LogInPresenterTest.java [hunk 18]
```diff
@@ -1,3 +1,36 @@
-package ar.com.wolox.android.training;
+package ar.com.wolox.android.training.ui.login;
+import static org.assertj.core.api.Assertions.*;
+//import static org.mockito.ArgumentMatchers.any;
```
Comment (alvarez-fabian) [inline app/src/test/java/ar/com/wolox/android/training/ui/login/LogInPresenterTest.java]: Remove commented lines :muscle:

File: app/src/test/java/ar/com/wolox/android/training/ui/login/LogInPresenterTest.java [hunk 19]
```diff
@@ -1,3 +1,36 @@
...
+    RetrofitServices retrofitServicesMock;
+    LogInService logInServiceMock;
+    LogInPresenter logInPresenter;
+
+    @Before
+    public void beforeTest(){
+        retrofitServicesMock = Mockito.mock(RetrofitServices.class);
+        logInServiceMock = Mockito.mock(LogInService.class);
+        Mockito.doReturn(logInServiceMock).when(retrofitServicesMock).getService(eq(LogInService.class));
+        //Mockito.when(retrofitServicesMock.getService(eq(LogInService.class))).thenReturn(logInServiceMock);
```
Comment (alvarez-fabian) [inline app/src/test/java/ar/com/wolox/android/training/ui/login/LogInPresenterTest.java:27]: Here too

File: app/src/test/java/ar/com/wolox/android/training/ui/login/LogInPresenterTest.java [hunk 20]
```diff
@@ -1,3 +1,36 @@
...
+        logInPresenter = new LogInPresenter(null, retrofitServicesMock);
+    }
+
+    @Test
+    public void sarasa(){
+        logInPresenter.doLogin("sadsad", "afrafea");
+        assertThat(4).isLessThan(3);}
```
Comment (alvarez-fabian) [inline app/src/test/java/ar/com/wolox/android/training/ui/login/LogInPresenterTest.java:34]: Put the curly brace in a new line :smile:

**Review by alvarez-fabian — CHANGES_REQUESTED** [review]
Just a couple of style fixes. Almost done!

**Comment by lmiotti** [comment]
Commented lines were removed and curly brace is now in a new line.

**Pull request closed without merging**